### DISCUSSION TITLE
Erasure configuration

### DISCUSSION THREAD

**Discussion by psxReboot** [minio/minio#18265] [Q&A]
Hello.
I can’t find an example of setting the erasure code parity (M) and erasure code stripe size (K+M).
If I select M=2 and K+M=8, where can I configure this?
Maybe a link to the documentation?

**Comment by klauspost** [comment]
Generally you shouldn't manually configure K+M.

Parity shards can easily be configured with `mc admin config set ALIAS storage_class standard=EC:2` or via env vars: https://min.io/docs/minio/linux/reference/minio-server/minio-server.html#storage-class

**Comment by zig** [comment]
In the example of 2 servers, 2 drives per server, K+M = 4 by default, but how is chosen M ? Will it be 1 or 2 ? Depending on this, the usable capacity and drive failure tolerance will be very different. Where can one see the chosen M parameter ?

**Reply by marktheunissen** [reply]
You can run `mc admin info ALIAS` and you'll see `EC:1` or `EC:2` on the last line of the output

**Reply by AlexZIX** [reply]
> You can run `mc admin info ALIAS` and you'll see `EC:1` or `EC:2` on the last line of the output

I can't find the EC:X in mc admin info ALIAS output.

**Reply by AlexZIX** [reply]
> In the example of 2 servers, 2 drives per server, K+M = 4 by default, but how is chosen M ? Will it be 1 or 2 ? Depending on this, the usable capacity and drive failure tolerance will be very different. Where can one see the chosen M parameter ?

As klauspost said you shouldn't manually configure K or M or K+M. In you case setting EC:2 will means that 2 disks will be dedicate for data and 2 for parity, EC:1 means 3 for data and 1 for parity. In EC:X variable X means how much disks will be dedicated for parity. It can be from 0 to 1/2 of disks in your erasure set.
